src/event/event_set.rs

```diff
@@ -63,63 +21,44 @@ type EventStateList = Vec<EventState, { EventManager::EVENT_COUNT }>;
 pub(crate) struct EventSet {
     /// Tasklet assigned to this set.
     tasklet: TaskletPtr,
-    /// List of event states.
-    event_states: Mutex<EventStateList>,
+    /// Event queue.
+    event_queue: Mutex<EventQueue>,
 }
 
 impl EventSet {
     /// Creates new event set.
     pub(crate) fn new(tasklet: TaskletPtr) -> Self {
         EventSet {
             tasklet,
-            event_states: EventStateList::new().into(),
+            event_queue: EventQueue::new().into(),
         }
     }
 
-    /// Adds new event to the set.
-    ///
-    /// # Parameters
-    /// * `event_id` - ID of the event that is to be added to this set.
-    ///
-    /// # Return
-    /// `()` if successful, `InitError` otherwise.
-    ///
-    /// # Safety
-    /// This is unsafe, because it mutably borrows the list of event states.
-    /// This is safe to call before the system initialization.
-    pub(crate) unsafe fn add_event(&self, event_id: EventId) -> Result<(), InitError> {
-        let event_state = EventState::new(event_id);
-
-        self.event_states
-            .lock(|event_states| match event_states.push(event_state) {
-                Ok(_) => Ok(()),
-                Err(_) => Err(InitError::EventSetFull),
-            })
-    }
-
     /// Activates event
     ///
     /// # Parameters
     /// * `event_id` - Event ID to activate.
     ///
     /// # Return
     /// `()` if successful, `RuntimeError` otherwise.
```

Comment:
```suggestion
    /// `true` if successfully activated event, `false` if event was already on the event queue and is waiting for trigger, `RuntimeError` otherwise.
```

Reply:
if that description is incorrect, please fix, would be nice to know when exactly this function can fail

Reply:
I planned to document what errors can be returned from the functions, but I dropped that idea. It would require to much of manual checking to keep documentation 100% up-to-date and correct.

Reply:
yeah, there's no need to document every possible error, an explanation "when it's safe to assume that this function can fail" is good enough